python/lsst/daf/butler/transfers/_yaml.py

```diff
@@ -64,6 +64,8 @@
 this version of the code.
 """
 
+_refIntId2UUID = defaultdict[int, uuid.UUID](uuid.uuid4)
```

Comment:
Is this a workaround for input YAML files that still have integer dataset IDs in them? Should we instead fix those YAML files? It may also be better to generate reproducible UUIDs in that case if you want to keep this map.

Reply:
I could not be sure all the places where this might be happening, so I opted to fix it here. Good point on the reproducibility I will go with that.